pkg/command/context_test.go

```diff
@@ -449,6 +450,92 @@ var _ = Describe("create new context", func() {
 			})
 		})
 
+		Describe("create context with self-managed tmc endpoint", func() {
+			var (
+				tkgConfigFile   *os.File
+				tkgConfigFileNG *os.File
+				err             error
+				ctx             *configtypes.Context
+			)
+
+			BeforeEach(func() {
+				tkgConfigFile, err = os.CreateTemp("", "config")
+				Expect(err).To(BeNil())
+				err = copy.Copy(filepath.Join("..", "fakes", "config", "tanzu_config.yaml"), tkgConfigFile.Name())
+				Expect(err).To(BeNil(), "Error while copying tanzu config file for testing")
+				os.Setenv("TANZU_CONFIG", tkgConfigFile.Name())
+
+				tkgConfigFileNG, err = os.CreateTemp("", "config_ng")
+				Expect(err).To(BeNil())
+				os.Setenv("TANZU_CONFIG_NEXT_GEN", tkgConfigFileNG.Name())
+				err = copy.Copy(filepath.Join("..", "fakes", "config", "tanzu_config_ng.yaml"), tkgConfigFileNG.Name())
+				Expect(err).To(BeNil(), "Error while coping tanzu config file for testing")
+			})
+			AfterEach(func() {
+				os.Unsetenv("TANZU_CONFIG")
+				os.Unsetenv("TANZU_CONFIG_NEXT_GEN")
+				os.RemoveAll(tkgConfigFile.Name())
+				os.RemoveAll(tkgConfigFileNG.Name())
+				resetContextCommandFlags()
+			})
+			Context("with endpoint and context name provided", func() {
+				It("should create context with given endpoint and context name", func() {
+					selfManaged = true
+					endpoint = fakeTMCEndpoint
+					ctxName = testContextName
+					ctx, err = createNewContext()
+					Expect(err).To(BeNil())
+					Expect(ctx.Name).To(ContainSubstring("fake-context-name"))
+					Expect(string(ctx.Target)).To(ContainSubstring("mission-control"))
+					Expect(ctx.GlobalOpts.Endpoint).To(ContainSubstring(endpoint))
+				})
+			})
+			Context("context name already exists", func() {
+				It("should return error", func() {
+					selfManaged = true
+					endpoint = fakeTMCEndpoint
+					ctxName = exisitingContext
```

Comment:
Do you want to take this opportunity to fix the global typo in `exisitingContext` to `existingContext`?

Reply:
fixed.thanks